### PR TITLE
feat(zetacore): add chain static information for bitcoin signet testnet

### PR DESCRIPTION
**Pull request by kingpinXD**
# Description

Closes : https://github.com/zeta-chain/node/issues/2884

# How Has This Been Tested?

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions

## Summary by CodeRabbit

- **New Features**
	- Added support for the Bitcoin Signet Testnet, enhancing documentation and API capabilities.
	- Introduced `btc_signet_testnet` to the list of valid chain names in the API.
	- Updated configurations to include Bitcoin Signet Testnet for better management and usability.

- **Bug Fixes**
	- Improved clarity and maintainability of Bitcoin chain parameters and configurations.

- **Documentation**
	- Enhanced changelog and API documentation to reflect new functionalities related to Bitcoin Signet Testnet.

### CODE REVIEW

File: pkg/chains/chains.go [hunk 1]
```diff
@@ -169,6 +169,18 @@ var (
...
+	BitcoinSignetTestnet = Chain{
+		ChainName:   ChainName_btc_signet_testnet,
+		ChainId:     001,
+		Network:     Network_btc,
+		NetworkType: NetworkType_testnet,
+		Vm:          Vm_no_vm,
+		Consensus:   Consensus_bitcoin,
+		IsExternal:  true,
+		CctxGateway: CCTXGateway_observers,
+		Name:        "btc_signet_testnet",
```
Comment (ws4charlie) [inline pkg/chains/chains.go:181]: maybe `btc_signet` to be consistent?

File: proto/zetachain/zetacore/pkg/chains/chains.proto [hunk 2]
```diff
@@ -48,6 +48,8 @@ enum ChainName {
   solana_mainnet = 21;
   solana_devnet = 22;
   solana_localnet = 23;
+
+  btc_signet_testnet = 24;
```
Comment (ws4charlie) [inline proto/zetachain/zetacore/pkg/chains/chains.proto:52]: same

**Review by ws4charlie — APPROVED** [review]
overall looks good. minor comment

File: proto/zetachain/zetacore/pkg/chains/chains.proto (hunk 2 above)
Comment (kingpinXD) [reply proto/zetachain/zetacore/pkg/chains/chains.proto:52]: I considered that , but wanted to keep `testnet` in the name to be consistent with other networks, which follow the name patters network_netoworkType

**Pull request merged**